### ISSUE TITLE
Gamma is not decreasing

### ISSUE THREAD

**Issue by NinaMaz** [oval-group/dfw#6]
Hello! I have faced an issue similar to #3: no matter the value of the initial learning rate, my gamma is always increasing until it hits some high value and oscillates around it (in my case it is definitely bigger than 1, varying from 400 to 2000 depending on eta). 

I also tried to increase the batch size as it is suggested in the article but the results stay the same.

Initially I train my model for 1000 epochs with the learning rate set to 0.001 and batch size to 32 with some simple learning rate schedule using Adam optimizer. 

Do you have any suggestions?

**Comment by lberrada** [comment]
Hi, thanks for the detailed issue.

Have you tried de-activating momentum if you are using it? We have found out that using momentum can improve generalization but is sometimes detrimental to the quality of the optimization (momentum was originally designed for non-stochastic optimization).

If that does not solve the issue, note that DFW also supports a manual schedule for the learning rate and that it should require less tuning than SGD. For instance, I would try a schedule like `eta` for the first half of the training, and `0.1 * eta` for the second half (where `eta` would be cross-validated).

In case you are interested in more details about conditions of convergence, a current work in progress hints that an algorithm like DFW may only converge when the overall objective is driven close to its global minimum. This is analogous to [SGD needing over-parameterization to converge with a constant step-size](https://arxiv.org/abs/1810.07288). Therefore, if the model is not able to drive the overall objective close to zero, the DFW algorithm may not be able to converge with a constant value for the learning rate `eta` (in which case `eta` needs to be decayed during training).
